### ISSUE TITLE
Implement `Account` entity and its table

### ISSUE THREAD

**Issue by nazrinharris** [nazrinharris/pecunia#16]
Not sure how I'm gonna go about with this. We ballin.

## Deed
- [x] Setup `Account` entity
- [x] Setup `AccountRepo` barebones
- [x] Setup `AccountLocalDS` barebones
- [x] Setup `AccountFailure`s 

## Actual Implementation
- [x] Do the actual implementation for `createAccount()`

## Add needed fields for DebugLocalDB
- [x] Setup form and business logic for creating an `Account`

## Time for `getAllAccounts()` and `watchAllAccounts()`
- [x] Do the implementation for the repo, local_ds, and dao
- [x] Do the accounts list UI in debug local db

**Comment by nazrinharris** [comment]
Cool. Got quite a lot done. Managed to fully implement create account. Now I should implement `getAllAccounts()` and `watchAllAccounts()`.

There's some worry that I have when it comes to currency. SInce this is going to be a multi-currency app, I'm not sure how that is going to work out with everything. On the other hand, I did find some pretty useful packages to work with:
1. https://pub.dev/packages/country_currency_pickers
2. https://pub.dev/packages/currency_formatter
